.github/workflows/publish-crates.yml

```diff
@@ -0,0 +1,79 @@
+#
+# Copyright (c) 2022, 2025 ZettaScale Technolgies.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+# which is available at https://www.apache.org/licenses/LICENSE-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+#
+# Contributors:
+#   ZettaScale Zenoh team, <zenoh@zettascale.tech>
+#
+name: Publish crates.io
+on:
+  push:
+    tags:
+      - 'v*'
+  workflow_dispatch:
+    inputs:
+      dryrun:
+        default: false
+        type: boolean
+
+env:
+  CARGO_REGISTRY_TOKEN: ${{ secrets.ZETTASCALE_BOT_CRATES_IO_TOKEN }}
+  CARGO_REGISTRY_CREDENTIAL_PROVIDER: "cargo:token"
+
+jobs:
+  build:
+
+    name: Build on ${{ matrix.os }}
+    runs-on: ["ubuntu-runner-set"]
+    strategy:
+      fail-fast: true
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        ref: ${{ github.ref }}
+
+    - name: Install toml-cli2
+      uses: taiki-e/cache-cargo-install-action@v2
+      with:
+        tool: toml-cli2
+      env:
+        CARGO_REGISTRY_DEFAULT: "crates-io"
+
+    - name: Install Rust toolchain
+    # NOTE: Showing the active Rust toolchain (defined by the rust-toolchain.toml file)
+    # will have the side effect of installing it; if it's not installed already.
+      run: |
+        rustup toolchain uninstall $(toml get rust-toolchain.toml toolchain.channel -r)
+        rustup toolchain install $(toml get rust-toolchain.toml toolchain.channel -r)
+        rustup component add rustfmt clippy
+
+    - name: Code format check
+      run: cargo fmt --check
+
+    - name: Clippy
+      run: cargo clippy --all-targets -- -D warnings
+
+    - name: Build
+      run: cargo build --verbose --all-targets
+
+    - name: Run tests
+      run: cargo test --verbose --release
+
+    - name: Cargo check
+      run: |
+        cargo check -p zrpc
+        cargo check -p zrpc-derive
+
+    - name: Publish
+      if: ${{ !inputs.dryrun }}
```

Comment:
I think we should add a check for the github events instead `if: ${{ github.event == "push" || (github.event == "workflow_dispatch" && inputs.dryrun == false) }}` that way you don't have to rely on the inputs that don't exist on a push event. When this is called manually via workflow dispatch, it'll check the value of the inputs.dryrun and run accordingly